### PR TITLE
Calendar model

### PR DESCRIPTION
**Pull request by tymmesyde**

### CODE REVIEW

File: src/models/calendar.rs
Comment (elpiel) [inline src/models/calendar.rs]: This is confusing. NaiveDate also has a day. Why is it different that the DateTime? 

- [x] Please handle  Day vs Month values differently

**Comment by TRtomasz** [comment]
imo no point of making a pr that does not include all needed dependencies to run. It is untestable. I this case it  should include in `package.json` core-web version that is still in PR without waiting for it to be merged.  Preferred solution would be to make this pr only after core and core-web changes are merged so we could avoid testing something without a reason in case it is decided that core/core-web changes should be rewritten or changed in a bigger part.

**Comment by elpiel** [comment]
There are a few things in the PR that I would like to change. Especially with handling the Date changes and date changes with offset. I've worked on some of the changes but left it as I had to work on more pressing issues.

Will get back to this.

File: src/models/calendar.rs
Comment (elpiel) [inline src/models/calendar.rs:103]: is this first_weekday indexed from 0 or from 1?
This usually can easily by extracted from methods on Date or DateTime instances, including how many days a certain month has.

File: src/models/calendar.rs
Comment (elpiel) [inline src/models/calendar.rs]: This is not how you should do it, especially the while loops..
There are methods that do this for you.

**Review by elpiel — CHANGES_REQUESTED** [review]

**Comment by elpiel** [comment]
I will still test the stremio-web for the UI implementation and general workings of the logic but there are a few things that need refactoring from the model.
Logic looks good so far though.

File: src/models/calendar.rs
Comment (elpiel) [inline src/models/calendar.rs]:
```suggestion
            let first_day_of_month = NaiveDate::from_ymd_opt(*year, *month, 1).unwrap_or_default();
            let first_of_next_month = first_day_of_month + Months::new(1);
            let last_day_of_month = first_of_next_month.pred_opt().unwrap_or_default();

            let days = last_day_of_month.day();

            let first_weekday = first_day_of_month.weekday().num_days_from_monday();
```

File: src/models/calendar.rs
Comment (elpiel) [inline src/models/calendar.rs]:
```suggestion
            let prev_date = date - Months::new(1);
            let next_date = date + Months::new(1);
```

File: src/models/calendar.rs
Comment (tymmesyde) [reply src/models/calendar.rs:103]: From 0
It's using `.weekday().num_days_from_monday()` from NaiveDate 
https://github.com/Stremio/stremio-core/pull/697/files#diff-2e83cb7dcf42829873903acb76fb7f66c5b91ddde7ab289eaf45f2013e788048R207

**Review by elpiel — APPROVED** [review]

**Pull request merged**
